src/Hypsography/Hypsography.jl

```diff
@@ -156,7 +160,45 @@ function ExtrudedFiniteDifferenceSpace(
     )
 end
 
-
+"""
+    diffuse_surface_elevation!(f::Field; κ::T, iter::Int, dt::T)
+
+Option for 2nd order diffusive smoothing of generated terrain.
+Mutate (smooth) a given elevation profile `f` before assigning the surface
+elevation to the `HypsographyAdaption` type. A spectral second-order diffusion 
+operator is applied with forward-Euler updates to generate
+profiles for each new iteration. Steps to generate smoothed terrain (
+represented as a ClimaCore Field) are as follows: 
+- Compute discrete elevation profile f
+- Compute diffuse_surface_elevation!(f, κ, iter). f is mutated.
+- Define `Hypsography.LinearAdaption(f)`
+- Define `ExtrudedFiniteDifferenceSpace` with new surface elevation.
+Default diffusion parameters are appropriate for spherical arrangements. 
+For `zmax-zsfc` == 𝒪(10^4), κ == 𝒪(10^8), dt == 𝒪(10⁻¹).
+"""
+function diffuse_surface_elevation!(
+    f::Fields.Field;
+    κ::T = 1e8,
+    maxiter::Int = 100,
+    dt::T = 1e-1,
+) where {T}
+    # Define required ops
+    wdiv = Operators.WeakDivergence()
+    grad = Operators.Gradient()
+    FT = eltype(f)
+    # Create dss buffer
+    ghost_buffer = (bf = Spaces.create_dss_buffer(f),)
+    # Apply smoothing
+    for iter in 1:maxiter
+        # Euler steps
+        χf = @. wdiv(grad(f))
+        Spaces.weighted_dss!(χf, ghost_buffer.bf)
+        @. f += κ * dt * χf
+        @. f = ifelse(f < FT(0), FT(0), f)
```

Comment:
Do we need this? Depressions are also valid topographical features right?

Reply:
Fro source files where we ignore bathymetric representations; this should be handled at the ClimaAtmos level; so this is unnecessary here. Removed and committed.